### PR TITLE
CacheDictionaryUpdateQueueSize metric added

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Now you can monitor the size of update queue in CacheDictionaries from system.metric_log.

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
Test will be added soon.

**Comment by nikitamikhaylov** [comment]
I looked through our test and didn't find any test for CurrentMetrics. :tired_face:

**Comment by alexey-milovidov** [comment]
> I looked through our test and didn't find any test for CurrentMetrics.

Global values of CurrentMetrics were difficult to test, because we have to check it at the right time.
Now it's possible to use `system.metric_log`.

(I'm not sure the test will worth adding)

File: dbms/src/Dictionaries/CacheDictionary.h [hunk 1]
```diff
@@ -364,6 +369,9 @@ class CacheDictionary final : public IDictionary
 
         std::atomic<bool> is_done{false};
         std::exception_ptr current_exception{nullptr};
+
+        /// While UpdateUnit is alive, it is accounted in update_queue size.
+        CurrentMetrics::Increment alive_update_unit{CurrentMetrics::CacheDictionaryUpdateQueueSize};
```
Comment (alexey-milovidov) [inline dbms/src/Dictionaries/CacheDictionary.h]: Why don't also add a metric with the number of keys?

Comment (alexey-milovidov) [reply dbms/src/Dictionaries/CacheDictionary.h]: And we can rename the metrics accordingly (batches/keys).

Comment (nikitamikhaylov) [reply dbms/src/Dictionaries/CacheDictionary.h]: Will do.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**